src/ethereum/EnigmaContractProductionWriterAPI.js

```diff
@@ -48,7 +48,7 @@ class EnigmaContractProductionWriterAPI extends EnigmaContractWriterAPI {
             reject(error);
           })
           .on(ETHEREUM_RECEIPT_EVENT, async receipt => {
-            if (!this.minimumConfirmations) {
+            if (this.minimumConfirmations == 0 || !Number.isInteger(minimumConfirmations)) {
```

Comment:
Isn't `(this.minimumConfirmations === 0)` better?
`minimumConfirmations` is undefined here (vs this.minimumConfirmations) and I am sure that this should be checked here and not when setting it...

Reply:
You can set it in multiple places (cli, builder, api).    
As I see it as long as this is a zero or not an integer the condition should be met. I'll add `===` but I don't think it matters here.